src/node/rpc/node_frontend.h

```diff
@@ -958,6 +964,75 @@ namespace ccf
           "status", ccf::endpoints::OptionalParameter)
         .install();
 
+      auto get_removable_nodes = [this](auto& args, nlohmann::json&&) {
+        GetNodes::Out out;
+
+        auto nodes = args.tx.ro(this->network.nodes);
+        nodes->foreach(
+          [this, &out, nodes](const NodeId& node_id, const NodeInfo& ni) {
+            auto node = nodes->get_globally_committed(node_id);
```

Comment:
Perhaps add a comment that this endpoint should be modified with caution as it checks for globally committed state and may break serialisation?

Reply:
I mean, sure, but then I feel this is true of many node frontend endpoints, chief among which `join`.

This isn't so much breaking serialization as it is adding a read dependency on something in addition to the KV state (the replication status). The transaction will still be coherent with the total order if it happens, but may not take place depending on extra-non-KV-state (same as trying to use any endpoint before open!).

Reply:
I think this still warrants a comment. It's a near-unique use of `get_globally_committed` (the only other being the frontend open check, which is at least far enough from any other KV operations in code-distance that it avoids a lot of potential confusion. We've convinced ourselves this non-transactional dependency is sound (as you outlined), but we should include that in a comment in the code for posterity + discoverability. I don't think `join` or the other node frontend endpoints do anything like this with non-transactional dependencies?

Reply:
`join` does redirect things no other endpoint does, many of the built-ins end up being fairly special, and I think we're on average better off with explicit naming and clear code than large comments that don't necessarily stay up to date. But I also believe in code reviews, so I'll add a comment :)

Reply:
@jumaffre @eddyashton I've re-read all three uses carefully, and concluded one of them was unnecessary. I've commented the other two.